x-pack/packages/kbn-elastic-assistant/impl/assistant/api.test.tsx

```diff
@@ -133,6 +133,28 @@ describe('API tests', () => {
       );
     });
 
+    it('calls the actions connector api with invoke when assistantStreamingEnabled is true when assistantLangChain is false and alerts is true', async () => {
+      const testProps: FetchConnectorExecuteAction = {
+        ...fetchConnectorArgs,
+        assistantLangChain: false,
+        alerts: true,
+      };
+
+      await fetchConnectorExecuteAction(testProps);
+
+      expect(mockHttp.fetch).toHaveBeenCalledWith(
+        '/internal/elastic_assistant/actions/connector/foo/_execute',
+        {
+          body: '{"params":{"subActionParams":{"model":"gpt-4","messages":[{"role":"user","content":"This is a test"}],"n":1,"stop":null,"temperature":0.2},"subAction":"invokeAI"},"assistantLangChain":false}',
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          signal: undefined,
+        }
+      );
+    });
```

Comment:
@stephmilovic would you be willing to confirm the observations in the `Summary of Observations` section are the expected behavior?

## Desk testing

This change was desk tested using the following combination of _Alerts_ and _Knowledge Base (ES|QL query generation)_ prompts:

```
Do I have open alerts?

What is the workflow status of my alerts?

What is the ID and details of the acknowledged alert? Answer using a markdown table

Generate an ES|QL query that will count the number of connections made to external IP addresses, broken down by user. If the count is greater than 100 for a specific user, add a new field called "follow_up" that contains a value of "true", otherwise, it should contain "false". The user names should also be enriched with their respective group names.
```

## Summary of Observations

- The `Stop streaming` option was ONLY observed with BOTH the KB and Alerts toggles disabled
  - (The `Stop streaming` option was NOT observed when ONLY the KB toggle is enabled)
- Prompts for ES|QL query generation frequently do NOT return the expected results when ONLY the KB toggle is enabled
  - All other combinations of toggles consistently returned the expected results for ES|QL query generation
- The error reproduced in `main`, when ONLY alerts (+ streaming) is enabled, was NOT observed when running this branch (the error appears to be resolved by the PR)

### Detailed Observations (with Video)

- When both the KB and Alerts toggles are disabled, the `Stop streaming` option IS displayed, and the response appears to be streamed, per the video below:

<https://github.com/elastic/kibana/assets/4459398/6429657f-28ba-48f9-a093-8c2bb024b721>

- When just the KB is enabled, the `Stop streaming` option is NOT displayed for 3 out 3 requests, and the responses do NOT appear to be streamed.

1) Just KB (1 of 3)

<https://github.com/elastic/kibana/assets/4459398/8d567127-bb67-4c32-8618-a19adae735ff>

2) Just KB (2 of 3)

<https://github.com/elastic/kibana/assets/4459398/704a78d7-4eec-4d87-8201-568f785bf212>

3) Just KB (3 of 3)

<https://github.com/elastic/kibana/assets/4459398/3090baa0-a495-40fe-819b-ddcdcfd21141>

- When just the KB is enabled, 2 of 3 responses failed to generate the expected ES|QL response
  - With additional ad hock testing, the "just KB" scenarios appear to return unexpected results more than any other scenario

- When just Alerts are enabled, the following error was reproduced on `main` with the _streaming_ and _RAG on Alerts_ feature flags enabled (via the PR description):

```
xpack.securitySolution.enableExperimental: ['assistantRagOnAlerts', 'assistantStreamingEnabled']
```

and (only) alerts enabled in settings:

![repro_settings](https://github.com/elastic/kibana/assets/4459398/e97e01c6-bc3f-4526-9368-1d25b275c79c)

### Before

![before](https://github.com/elastic/kibana/assets/4459398/99878b22-2be9-4c6b-bc57-c74bc8a679d5)

- When just Alerts are enabled (after the fix), the `Stop streaming` option is NOT displayed for 3 out of 3 requests, and the responses do NOT appear to be streamed:

1) Just Alerts (1 of 3)

<https://github.com/elastic/kibana/assets/4459398/225b4ece-f1b9-470e-ab55-d2bdec78ffce>

2) Just Alerts (2 of 3)

<https://github.com/elastic/kibana/assets/4459398/bc86e7ac-3db7-4c73-91ba-4b381aaf9b70>

3) Just Alerts (3 of 3)

<https://github.com/elastic/kibana/assets/4459398/226c74b6-2521-43cb-bdd0-ab05edb003b4>

- With just Alerts enabled, 3 of 3 scenarios had the expected responses

- When BOTH the KB and alerts are enabled, the `Stop streaming` option is NOT displayed for 3 out 3 KB requests with KB (ES|QL query generation) prompts

- BOTH enabled, KB (1 of 3)

<https://github.com/elastic/kibana/assets/4459398/f18624a6-5d8f-495e-b303-4a47df27e315>

- BOTH enabled, KB (2 of 3)

<https://github.com/elastic/kibana/assets/4459398/af14ca5c-6c67-46f7-942f-26f4f915ad08>

- BOTH enabled, KB (3 of 3)

<https://github.com/elastic/kibana/assets/4459398/7c945290-404a-4eba-937d-cd35a083e165>

- When BOTH the KB and alerts are enabled, the `Stop streaming` option is NOT displayed for 3 out 3 KB requests with Alerts prompts

- BOTH enabled, Alerts (1 of 3)

<https://github.com/elastic/kibana/assets/4459398/cd9acab6-67e6-4a8b-923f-18c656f6efb0>

- BOTH enabled, Alerts (2 of 3)

<https://github.com/elastic/kibana/assets/4459398/6cb07ba0-27e6-44c9-a3b5-0b68cf1d720f>

- BOTH enabled, Alerts (3 of 3)

<https://github.com/elastic/kibana/assets/4459398/f4790d8d-2c78-4f83-a771-21ed97f8f1c2>

- With BOTH the KB and Alerts enabled, 6 of 6 scenarios had the expected responses

Reply:
> @stephmilovic would you be willing to confirm the observations in the `Summary of Observations` section are the expected behavior?

@andrew-goldstein thanks for the detailed message. It seems the streaming is working as expected. I'm not sure why ESQL is not working as expected with KB turned on. Streaming is off when KB is on regardless of the feature flag, so it would not be related.